### ISSUE TITLE
Enums: false-positive "No base classes allowed after `ReprEnum`" error

### ISSUE THREAD

**Issue by AlexWaygood** [python/mypy#12787]
**Bug Report**

Minimal repro on 3.11 ([playground link](https://mypy-play.net/?mypy=latest&python=3.11&flags=show-error-codes&gist=62551352af494b26b9d2c540cbf4dfac)):
```python
from enum import ReprEnum, Flag
class MyFlag(ReprEnum, Flag): ...
```

Minimal repro on <=3.10 ([playground link](https://mypy-play.net/?mypy=0.940&python=3.10&flags=show-error-codes&gist=ca82dddbe120822a0f6c7459b34784b7)):
```python
from enum import Enum, Flag
class ReprEnum(Enum): ...
class MyFlag(ReprEnum, Flag): ...
```

mypy output for both (with small line-number differences):

```
main.py:4: error: No base classes are allowed after "__main__.ReprEnum"  [misc]
```

**Expected Behavior**

No error should be reported. This use of multiple inheritance works fine at runtime on 3.10 and 3.11. In fact, in 3.11, the source code for the `enum` module in the stdlib uses [exactly this inheritance structure](https://github.com/python/cpython/blob/2e8f721c0f275d9d7c018b4a2a66f053cf34bb36/Lib/enum.py#L1574-L1577):

```python
class ReprEnum(Enum):
    """
    Only changes the repr(), leaving str() and format() to the mixed-in type.
    """

# Much further down...

class IntFlag(int, ReprEnum, Flag, boundary=EJECT):
    """
    Support for integer-based Flags
    """
```

**Your Environment**

I can reproduce this bug [on mypy 0.940+](https://mypy-play.net/?mypy=0.940&python=3.11&flags=show-error-codes&gist=ddb43d68672ad20d807c6312ebad2312) (0.940 was the first released version with @sobolevn's #12026, which introduced this error message).

**Comment by AlexWaygood** [comment]
Even simpler repro that doesn't involve `enum.Flag`:

```python
from enum import Enum

class Foo(Enum): ...
class Bar(Enum): ...
class Baz(Foo, Bar): ...
```

Works fine at runtime, but [mypy says](https://mypy-play.net/?mypy=latest&python=3.10&gist=c64feb26c039e72690a1f0dc620e7512):

```
main.py:5: error: No base classes are allowed after "__main__.Foo"
```
